### ISSUE TITLE
Particles in their own subdir show a non-descriptive error on Layout save

### ISSUE THREAD

**Issue by fcFn** [gantry/gantry5#3191]
If you try to put particles into their own subfolder e.g. `particles/my-particle`, Gantry will crash at `preg_match()` with `Unknown modifier: m` on layout save.

Steps to reproduce:

1. Follow the steps in [the docs](https://docs.gantry.org/gantry5/advanced/creating-a-new-particle).
2. Create a subdirectory, e.g. `/particles/my-particle` and put your particle files there.
3. Use the particle in the Layout editor and save the layout.

Expected behavior:

It should save the layout without throwing an exception.

Possible fixes:
* Escape the path to the particle so that `preg_match` won't throw.
* Mention in the docs that particle files should be in a flat directory and either make the error message more descriptive or do not show the particles inside subdirectories in the Layout editor.

**Comment by N8Solutions** [comment]
Just my 2-cents here @fcFn. The documentation doesn't discuss the ability to place a particle into it's own subfolder under the `TEMPLATE_DIR/custom/particles` folder so the expected behavior is that this would fail, not that it would work.

So you're proposed "fix" isn't actually a fix, but rather it is considered to be an **enhancement**. If the files for the particles were all under one folder, i.e. `.scss` `.yaml` `.twig` & `.js` (if needed) then I could understand this but since those files are all under separate folders, I don't find your proposed enhancement necessary.

Can you provide a good case for the reasoning behind wanting to do this? I'm honestly curious because under the `/particles` folder there are only 2 files per particle, the `.twig` and the `.yaml` files.

**Comment by fcFn** [comment]
@N8Solutions You're right. Perhaps it should just fail with a better message instead of what it does now, such as "Please make sure your particle files are inside the root of the particles directory" (this is on Grav by the way, I'll edit the post). Or maybe it should be mentioned that `particles` has to be a flat directory in the docs.
